api/scpca_portal/models/dataset.py

```diff
@@ -67,6 +78,105 @@ class Meta:
+                ).values_list("scpca_id", flat=True)
+                self.data[project.scpca_id][Modalities.SPATIAL] = samples.filter(
+                    modality=Modalities.SPATIAL
+                ).values_list("scpca_id", flat=True)
+
+    def populate_ccdl_data_sample(self, sample: Sample):
+        self.data[sample.project.scpca_id] = {
+            "merge_single_cell": False,
+            "includes_bulk": False,
+            Modalities.SINGLE_CELL: [],
+            Modalities.SPATIAL: [],
+        }
+
+        modality = self.ccdl_type.get("modality")
+        self.data[sample.project.scpca_id][modality].append(sample.scpca_id)
+
+    @property
+    def is_ccdl_exists(self) -> bool:
+        matching = Dataset.objects.filter(is_ccdl=True, ccdl_name=self.ccdl_name)
+        if not matching.exists():
+            return False
+
+        for dataset in matching:
+            if set(dataset.data.keys()) == set(self.data.keys()):
+                return True
+
+        return False
+
+    def validate_samples(self) -> bool:
+        for project_id, project_config in self.data.items():
+            project = Project.objects.filter(scpca_id=project_id).first()
+            if not project:
+                logger.error(f"{self} invalid: Project {project_id} doesn't exist.")
+                return False
+            for modality in [Modalities.SINGLE_CELL, Modalities.SPATIAL]:
+                for sample_id in project_config[modality]:
+                    sample = Sample.objects.filter(scpca_id=project_id).first()
+                    if not sample:
+                        logger.error(
+                            f"{self} invalid: "
+                            f"Sample {sample_id} of modality {modality} "
+                            f"does not exist in Project {project_id}."
+                        )
+                        return False
+                    if sample.project.scpca_id != project_id:
+                        logger.error(
+                            f"{self} invalid: "
+                            f"Sample {sample_id} does not belong to Project {project_id}."
+                        )
+                        return False
+                    if not sample.libraries:
+                        logger.error(
+                            f"{self} invalid: "
+                            f"Sample {sample_id} of Project {project_id} has no libraries."
+                        )
+                        return False
+                    if modality not in sample.modalities:
+                        logger.error(
+                            f"{self} invalid: "
+                            f"Sample {sample_id} of Project {project_id} "
+                            f"not of modality {modality}."
+                        )
+                        return False
+
+        return True
+
+    def validate_ccdl_dataset(self) -> bool:
+        return (not self.is_ccdl_exists) and self.validate_samples()
```

Comment:
```suggestion
    def should_process(self) -> bool:
        ``` returns true if the dataset doesnt exist in the DB or if the matching dataset's hash is different.
        for now just check if the data attribute matches```
        if dataset := self.existing_ccdl_datasets.first():
            return self.data != dataset.data
        else:
            return True
```